### PR TITLE
Support for Secret Network

### PR DESCRIPTION
**Pull request by toml01**
## Description

Support for Secret Network. New Cosmos-based chain.
Still waiting for you guys to merge the [trustwallet/assets PR](https://github.com/trustwallet/assets/pull/23150)

I'm pretty sure we've covered everything, and once the `assets` PR is merged, this PR should be ok to merge this as well.
Please let me know if anything else is needed on our end 🙂 

## How to test

Secret unit tests.
Get some SCRT and create transfer TX to another address.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [X] Create pull request as draft initially, unless its complete.
- [X] Add tests to cover changes as needed.
- [X] Update documentation as needed.
- [X] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [X] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -2344,6 +2344,38 @@
...
+    "id": "secret",
+    "name": "Secret",
+    "displayName": "Secret",
+    "coinId": 529,
+    "symbol": "SCRT",
+    "decimals": 6,
+    "blockchain": "Cosmos",
+    "derivation": [
+      {
+        "path": "m/44'/529'/0'/0/0"
```
Comment (Milerius) [inline registry.json:2357]: I'm not sure why the derivation path is different than cosmos, when I'm using keplr + ledger, I'm validating Secret transaction using the cosmos app, could you please clarify?

Comment (toml01) [reply registry.json:2357]: @Milerius Secret Network is using a different derivation path (like [Kava](https://github.com/trustwallet/wallet-core/blob/master/registry.json#L1184) and [Crypto.org](https://github.com/trustwallet/wallet-core/blob/master/registry.json#L2264) for that matter)
When you use Keplr (hot wallet), it uses the right path.
If you want to use ledger with the right derivation path, you would need to use the Secret Network Ledger app.

**Review by Milerius — COMMENTED** [review]
We recently support the following one: https://github.com/trustwallet/wallet-core/issues/2626

Do you plan to add any specific feature like: https://github.com/SecretFoundation/SNIPs/blob/master/SNIP-20.md ?

File: registry.json (hunk 1 above)
Comment (Milerius) [reply registry.json:2357]: > @Milerius Secret Network is using a different derivation path (like [Kava](https://github.com/trustwallet/wallet-core/blob/master/registry.json#L1184) and [Crypto.org](https://github.com/trustwallet/wallet-core/blob/master/registry.json#L2264) for that matter) When you use Keplr (hot wallet), it uses the right path. If you want to use ledger with the right derivation path, you would need to use the Secret Network Ledger app.

Thanks for the explanation

**Comment by Milerius** [comment]
> @Milerius Yes we would like to add SNIP20 tokens at some point. Not in the immediate future

Thanks for the clarification!

File: src/Coin.cpp [hunk 2]
```diff
@@ -105,6 +106,7 @@ Zilliqa::Entry zilliqaDP;
 Nervos::Entry NervosDP;
 Everscale::Entry EverscaleDP;
 Hedera::Entry HederaDP;
+Cosmos::Entry CosmosDP; // TODO remove if the blockchain already exists, or just remove this comment if not
```
Comment (Milerius) [inline src/Coin.cpp:109]: to remove

File: src/Coin.cpp [hunk 3]
```diff
@@ -56,6 +56,7 @@
 #include "Zcash/Entry.h"
 #include "Zilliqa/Entry.h"
 #include "Hedera/Entry.h"
+#include "Cosmos/Entry.h" // TODO remove if the blockchain already exists, or just remove this comment if not
```
Comment (Milerius) [inline src/Coin.cpp:59]: to remove

**Comment by toml01** [comment]
Also, I'm not sure why the SonarCloud test is failing. If there's anything I need to fix, please lmk

**Review by Milerius — CHANGES_REQUESTED** [review]

**Comment by Milerius** [comment]
> Also, I'm not sure why the SonarCloud test is failing. If there's anything I need to fix, please lmk

External Pull request can't access organisation tokens for security reason.

**Comment by toml01** [comment]
Removed both includes

**Review by hewigovens — APPROVED** [review]

**Review by catenocrypt — APPROVED** [review]

**Comment by Milerius** [comment]
@toml01 can you sync your fork with master branch, we pushed a fix that should disable SonarCloud for fork repository.

**Comment by toml01** [comment]
@Milerius Done. Need you guys to approve running workflows.
Also we're all set on the assets repo: https://github.com/trustwallet/assets/pull/23217

**Pull request merged**